buildvertical.targets

```diff
@@ -27,13 +27,21 @@
     </MSBuild>
 
     <ItemGroup>
-      <!-- transform back to project -->
-      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')">
-        <AdditionalProperties>Configuration=%(Identity);%(_ProjectBestConfigurations.AdditionalProperties)</AdditionalProperties>
+      <!-- transform back to project keeping the original item order. -->
+      <_ProjectBestConfigurations>
+        <Configuration>%(Identity)</Configuration>
+      </_ProjectBestConfigurations>
+      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')" />
+      <ProjectWithConfiguration>
+        <AdditionalProperties>Configuration=%(Configuration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
       </ProjectWithConfiguration>
     </ItemGroup>
 
     <ItemGroup>
+      <ProjectWithConfiguration RemoveMetadata="Configuration" />
```

Comment:
As discussed offline, Configuration metadata was only used as a temp variable to hold the configuration and later passed in as AdditionalProperties. This is getting re-engineered by @ericstj work on #15616